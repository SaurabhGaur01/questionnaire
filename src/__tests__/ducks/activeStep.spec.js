import {
    setActiveStep,
    activeStepReducer
} from '../../ducks/activeStep';

describe('activeStep Duck Tests', () => {
    const mockState = 1;

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setActiveStep should match the snapshot' , () => {
            expect(setActiveStep(mockState)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(activeStepReducer()).toEqual(0);
        });
        it('should return the passed state if called with no action' , () => {
            expect(activeStepReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(activeStepReducer(mockState, unknownAction)).toBe(mockState);
        });
    });

    describe('when called with a particular action', () => {
        it('should return the value which is passed' , () => {
            expect(activeStepReducer('', setActiveStep(mockState)))
            .toEqual(mockState);
        });
    });

})