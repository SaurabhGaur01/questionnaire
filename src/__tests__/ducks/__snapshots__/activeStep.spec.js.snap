// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeStep Duck Tests Action Generator setActiveStep should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_ACTIVE_STEP",
  "value": 1,
}
`;
